contracts/token/psp22/tests.rs

```diff
@@ -291,4 +297,34 @@ mod tests {
 
         psp22.transfer_from(accounts.alice, accounts.eve, alice_balance + 1, Vec::<u8>::new());
     }
+
+    /// The mint was applied.
+    #[ink::test]
+    fn mint_extension_works() {
+        let mut psp22 = PSP22Struct::new(100);
+        let old_supply = psp22.total_supply();
+        psp22.mint(old_supply);
+        let new_supply = psp22.total_supply();
+        assert_eq!(2 * old_supply, new_supply);
+    }
+
+    /// The burn supply was applied.
+    #[ink::test]
+    fn burn_extension_works() {
+        let mut psp22 = PSP22Struct::new(100);
+        let old_supply = psp22.total_supply();
+        psp22.burn(old_supply);
+        let new_supply = psp22.total_supply();
+        assert_eq!(old_supply - old_supply, new_supply);
+    }
+
+    /// The deposit_for was applied.
+    #[ink::test]
+    fn deposit_works() {
```

Comment:
In unit tests you can't test the logic with cross contract calls=) You can do that only with integration tests.

Reply:
Yeah, I know, just checked if everything okay with access to this function